Mathlib/Combinatorics/SimpleGraph/ConcreteColorings.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2023 Iván Renison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Iván Renison
+-/
+import Mathlib.Combinatorics.SimpleGraph.Coloring
+import Mathlib.Combinatorics.SimpleGraph.Hasse
+import Mathlib.Data.ZMod.Basic
+
+/-!
+# Concrete colorings of common graphs
+
+This file defines colorings for some common graphs
+
+## Main declarations
+
+* `SimpleGraph.pathGraph.bicoloring`: Bicoloring of a path graph
+-/
+
+theorem suc_mod2_neq (m : ℕ) : (m % 2 == 0) ≠ ((m + 1) % 2 == 0) := by
```

Comment:
In mathlib we use `=` instead of `==` and rely on the `decide` insertion.

This lemma made me realize we're missing the following lemmas. Could you add them?
```lean
-- to Data.Nat.Parity
theorem succ_mod_two_eq_zero_iff {m : ℕ} : (m + 1) % 2 = 0 ↔ m % 2 = 1 := by
  simp [← Nat.even_iff, ← Nat.not_even_iff, parity_simps]

theorem succ_mod_two_eq_one_iff {m : ℕ} : (m + 1) % 2 = 1 ↔ m % 2 = 0 := by
  simp [← Nat.even_iff, ← Nat.not_even_iff, parity_simps]
```

Style comments: The theorem you wrote might be called something like `mod_two_beq_zero_neq_succ_mod_two_eq_zero` using the mathlib naming convention, possibly shortened. It's `succ` instead of `suc` and `mod_two` instead of `mod2`.